src/lib/mongoBackend/mongoQueryTypes.cpp

```diff
@@ -434,24 +435,31 @@ HttpStatusCode mongoEntityTypes
         }
 
         /* Note that we need and extra query() to the database (inside attributeType() function) to get each attribute type.
-         * This could be unefficient, specially if the number of attributes is large */
-        std::vector<std::string> attrTypes;
-        getAttributeTypes(tenant, servicePathV, entityType->type , attrsArray[jx].str(), &attrTypes);
-
-        for (unsigned int kx = 0; kx < attrTypes.size(); ++kx)
+         * This could be inefficient, especially if the number of attributes is large */
+        if (!noAttrDetail)
```

Comment:
Just to see if I'm understanding correctly the modification... You have move the all logic under an `if (!noAttrDetail) and, in addition, adding a else clause. Is that correct?